packages/grid/_modules_/grid/hooks/features/useColumnResize.tsx

```diff
@@ -110,12 +111,134 @@ export const useColumnResize = (columnsRef: React.RefObject<HTMLDivElement>, api
     doc.addEventListener('mouseup', handleResizeMouseUp);
   });
 
+  // TODO: remove support for Safari < 13.
+  // https://caniuse.com/#search=touch-action
+  //
+  // Safari, on iOS, supports touch action since v13.
+  // Over 80% of the iOS phones are compatible
+  // in August 2020.
+  let cachedSupportsTouchActionNone;
+  function doesSupportTouchActionNone() {
```

Comment:
is it pure function? Can we move it to a utils file to keep the hook clean

Reply:
It's not pure, there is a cache.

I definitely agree that we should isolate the logic, so moving it to the top scope would be much better.

Regarding creating a utility file for it, why about we wait for the need to share the logic? This way we can more easily refactor in the future. It's not like having a dedicated file really help to avoid duplication or with discoverability, it doesn't much.

Reply:
I think if it can be outside the component, it's better to move it.
About the extra file, it's just to keep this file clean and focus on its own purpose. Creating another file is not only to share code, it can also be to keep the code clean... Up to you, I don't have a strong opinion on this, as the file is not that big.

Reply:
It's not a pure function but I will rework it because I took it from the slider directly and wanted to validate the approach first. I'll convert it to a memo and the `cachedSupportsTouchActionNone ` to a ref so it's more optimal

Reply:
What do you mean by keeping code clean? Why is it cleaner? From what I understand, we could have files with 1,000 LOCs. I think that the most important aspect to optimize on this matter is: How easy can we delete code? If you can delete, you can refactor, you can't understand what's going on either.

Reply:
I don't get it, why does it need a memo? I can be in the outer scope, it doesn't need to be in the scope of the hook.

Reply:
Ok, I've moved these 2 at the top of the file so in the future when we need it we can easily abstract them in their own utility file.